src/mono/browser/runtime/runtime.c

```diff
@@ -356,3 +357,25 @@ mono_wasm_assembly_find_method (MonoClass *klass, const char *name, int argument
 	MONO_EXIT_GC_UNSAFE;
 	return result;
 }
+
+/*
+ * mono_wasm_marshal_get_managed_wrapper:
+ * Creates a wrapper for a function pointer to a method marked with
+ * UnamangedCallersOnlyAttribute.
+ * This wrapper ensures that the interpreter initializes the pointers.
+ */
+void
+mono_wasm_marshal_get_managed_wrapper (const char* assemblyName, const char* nspace, const char* typeName, const char* methodName, int num_params)
+{
+	MonoError error;
+	mono_error_init (&error);
+	MonoAssembly* assembly = mono_wasm_assembly_load (assemblyName);
+	assert (assembly);
+	MonoClass* class = mono_wasm_assembly_find_class (assembly, nspace, typeName);
```

Comment:
Is the reflection here in conflict with our (Native)AOT goals ?

Reply:
> Is the reflection here in conflict with our (Native)AOT goals ?

This is an interpreter issue, the code in question already works in NativeAOT